src/components/ui/toolbox.ts

```diff
@@ -90,28 +108,51 @@ export default class Toolbox extends Module<ToolboxNodes> {
    * Tooltip utility Instance
    */
   private tooltip: Tooltip;
+
   /**
-   * @class
-   * @param {object} moduleConfiguration - Module Configuration
-   * @param {EditorConfig} moduleConfiguration.config - Editor's config
-   * @param {EventsDispatcher} moduleConfiguration.eventsDispatcher - Editor's event dispatcher
+   * Id of listener added used to remove it on destroy()
    */
-  constructor({ config, eventsDispatcher }: ModuleConfig) {
-    super({
-      config,
-      eventsDispatcher,
-    });
+  private clickListenerId: string = null;
+
+  /**
+   * Toolbox constructor
+   *
+   * @param options - available parameters
+   * @param options.api - Editor API methods
+   * @param options.tools - Tools available to check whether some of them should be displayed at the Toolbox or not
+   * @param options.shortcutsScopeElement - parent element for Shortcuts scope restriction
+   * @param options.onOpen - opening callback
+   * @param options.onClose - closing callback
+   * @param options.onBlockAdded - block adding callback
+   */
+  constructor({ api, tools, shortcutsScopeElement, onOpen, onClose, onBlockAdded }) {
+    this.api = api;
+    this.tools = tools;
+    this.onOpen = onOpen;
+    this.onClose = onClose;
+    this.onBlockAdded = onBlockAdded;
```

Comment:
In case API would grow, might be better to just extend EventEmitter interface and provide enum of possible events

```
toolbox.on(ToolboxEvents.Close, () => {});
toolbox.on(ToolboxEvents.Open, () => {});
```

Using typescript we can strictly type that

Reply:
done